guides/common/modules/proc_decreasing-performance-impact-of-the-pull-based-rex-transport.adoc

```diff
@@ -0,0 +1,31 @@
+[id="Decreasing_performance_impact_of_the_pull-based_REX_transport_{context}"]
+= Decreasing Performance Impact of the Pull-Based REX Transport
+
+When {ProjectServer} is configured to use REX pull-based transport for remote execution jobs using "Script" provider, {SmartProxyServer} sends a notification about new job to the client machines via MQTT.
+This notification does not include actual workload (actual script) the client is supposed to execute.
+Once a client receives a message about a new remote execution job, it queries the {SmartProxyServer} for actual payload.
+As the client executes the job, it periodically sends outputs of the job to the capsule,further increasing the number of requests to {SmartProxyServer}.
+
+These requests to {SmartProxyServer} together with high concurrency allowed by MQTT protocol can cause exhaustion of {SmartProxyServer} available connections and some requests might fail, making some child tasks of remote execution jobs to hang. 
+This also depends on actual job payload, as some jobs are causing additional load to {ProjectServer}, making it compete for resources if you are using integrated {SmartProxyServer}.
+
+To avoid this, configure your {ProjectServer} and {SmartProxyServer} deployment appropriately with the following parameters:
+
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+* MQTT Time To Live {endash} Time interval in seconds given to the host to pick up the job before considering the job undelivered
+* MQTT Rate Limit {endash} Number of jobs that are allowed to run at the same time
+* MQTT Resend Interval {endash} Time interval in seconds at which the notification should be re-sent to the host until the job is picked up or cancelled
+----
```

Comment:
```suggestion
MQTT Time To Live::
Time interval in seconds given to the host to pick up the job before considering the job undelivered
MQTT Rate Limit::
Number of jobs that are allowed to run at the same time
MQTT Resend Interval::
Time interval in seconds at which the notification should be re-sent to the host until the job is picked up or cancelled
```

alternative suggestion to show this. I am unsure about the bullet points within a code block.

Reply:
Hello, thank you for your suggestion. I am working on the performance testing for these values. Will edit or update this in next coming week.